src/components/Hoverable/index.js

```diff
@@ -1,189 +1,181 @@
 import _ from 'underscore';
-import React, {Component} from 'react';
+import React, {useEffect, useCallback, useState, useRef, useMemo, useImperativeHandle} from 'react';
 import {DeviceEventEmitter} from 'react-native';
 import {propTypes, defaultProps} from './hoverablePropTypes';
 import * as DeviceCapabilities from '../../libs/DeviceCapabilities';
 import CONST from '../../CONST';
 
+function mapChildren(children, callbackParam) {
+    if (_.isArray(children) && children.length === 1) {
+        return children[0];
+    }
+
+    if (_.isFunction(children)) {
+        return children(callbackParam);
+    }
+
+    return children;
+}
+
 /**
  * It is necessary to create a Hoverable component instead of relying solely on Pressable support for hover state,
  * because nesting Pressables causes issues where the hovered state of the child cannot be easily propagated to the
  * parent. https://github.com/necolas/react-native-web/issues/1875
  */
-class Hoverable extends Component {
-    constructor(props) {
-        super(props);
 
-        this.handleVisibilityChange = this.handleVisibilityChange.bind(this);
-        this.checkHover = this.checkHover.bind(this);
+function InnerHoverable({disabled, onHoverIn, onHoverOut, children, shouldHandleScroll}, outerRef) {
+    const [isHovered, setIsHovered] = useState(false);
+    const [isScrolling, setIsScrolling] = useState(false);
```

Comment:
I would suggest replacing `isScrolling` state in favour of `isScrollingRef`. Since using `isScrolling` state will re-render the `Hoverable` component and we can avoid it by using ref, which gives us a similar functionality but with less re-renders.

<img width="1311" alt="Screenshot 2023-09-26 at 1 46 12 PM" src="https://github.com/Expensify/App/assets/47336142/919aff74-13e0-4e98-b59e-37724faf8a97">

Reply:
Got you! I've implemented suggested changes 👍 Let's review it once again and we are good to go :)